From 6b12cd0281a75732fabe3b515b72354f5fd5015e Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 21 Jan 2015 13:27:24 +0800
Subject: [PATCH 0814/1594] MLK-10117-2:pxp: Remove runtime suspend/resume
 function

commit 2dce42037430b685836c0b149d185a51b0c3d503 from
git://git.freescale.com/imx/linux-2.6-imx.git

pxp module have involved to generic pm domain,
runtime_suspend/resume function will called when system
suspend/resume. It will cause request_bus_freq/release_bus_freq
called counter mismatch.
So move request_bus_freq/release_bus_freq to clk_enable/disable
function.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/dma/pxp/pxp_dma_v2.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index 12a98a1..37d4444 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -1074,6 +1074,8 @@ static void pxp_clk_enable(struct pxps *pxp)
 		return;
 	}
 
+	request_bus_freq(BUS_FREQ_HIGH);
+
 	pm_runtime_get_sync(pxp->dev);
 
 	if (pxp->clk_disp_axi)
@@ -1108,6 +1110,8 @@ static void pxp_clk_disable(struct pxps *pxp)
 	pm_runtime_put_sync_suspend(pxp->dev);
 
 	mutex_unlock(&pxp->clk_mutex);
+
+	release_bus_freq(BUS_FREQ_HIGH);
 }
 
 static inline void clkoff_callback(struct work_struct *w)
@@ -1789,17 +1793,13 @@ static int pxp_resume(struct device *dev)
 #ifdef CONFIG_PM_RUNTIME
 static int pxp_runtime_suspend(struct device *dev)
 {
-	release_bus_freq(BUS_FREQ_HIGH);
 	dev_dbg(dev, "pxp busfreq high release.\n");
-
 	return 0;
 }
 
 static int pxp_runtime_resume(struct device *dev)
 {
-	request_bus_freq(BUS_FREQ_HIGH);
 	dev_dbg(dev, "pxp busfreq high request.\n");
-
 	return 0;
 }
 #else
-- 
1.7.5.4

